internal/k8s/entity_test.go

```diff
@@ -173,6 +174,45 @@ func TestHasKind(t *testing.T) {
 	assert.True(t, svc.HasKind("service"))
 }
 
+func TestNamespacesAndCRDsFirst(t *testing.T) {
+	eDeploy := mustParseYAML(t, testyaml.DoggosDeploymentYaml)[0]
+	eService := mustParseYAML(t, testyaml.DoggosServiceYaml)[0]
+	eCRD := mustParseYAML(t, testyaml.CRDYAML)[0]
+	eNamespace := mustParseYAML(t, testyaml.MyNamespaceYAML)[0]
+
+	for _, test := range []struct {
+		name           string
+		input          []K8sEntity
+		expectedOutput []K8sEntity
+	}{
+		{"namespace first",
+			[]K8sEntity{eDeploy, eNamespace, eService},
```

Comment:
fuzz-testing is sometimes a more fun way to test these kinds of things. Just specify the correct input, call rand.Shuffle N times, and verify that the sort gets you back to the input.
https://golang.org/pkg/math/rand/#Shuffle

Reply:
yeah, I'd considered that! but i also wanted to ensure the behavior with e.g., only namespaces/crds, no namespaces/crds, etc.